core/src/main/scala/akka/persistence/jdbc/state/DurableStateQueries.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.jdbc.state
+
+import slick.jdbc.{ JdbcProfile, SetParameter }
+import akka.persistence.jdbc.config.DurableStateTableConfiguration
+
+class DurableStateQueries(val profile: JdbcProfile, override val durableStateTableCfg: DurableStateTableConfiguration)
+    extends DurableStateTables {
+  import profile.api._
+
+  lazy val sequenceNextValUpdater = durableStateTableCfg.schemaType match {
+    case "H2"       => new H2SequenceNextValUpdater(profile)
+    case "Postgres" => new PostgresSequenceNextValUpdater(profile)
+  }
```

Comment:
Are we planning to add the others or have we discovered that the trick we do with the seqNum is not possible with the other DBs? 

Otherwise, we can add a catch-all and throw an informative exception.

Reply:
We will add other dbs as well. I am sure the scheme will work for other dbs. But since we decided to test for H2 and Postgres for the time being, I kept it like that. Added a `case_ => ???` that will throw for now.